### PR TITLE
Doc updates for snake_case to camelCase

### PR DESCRIPTION
**Pull request by artberger**
# Description

Updated snake case to camel case for fields in Gloo resources. Method I took was:
1. Search doc folder for YAMLs with a solo.io API version `(^apiVersion).*(solo)` so I'd only find Gloo resources.
2. Search those YAML snippets for underscores to see snake_case fields.
3. If any snake cases were found, compare with the [CRDs](https://github.com/solo-io/gloo/tree/master/install/helm/gloo/crds) or the `-o yaml` output of the resource in my cluster to double-check that they should be camelCase.

# Context

See issue #5151. One thing to note is that the issue mentioned `lb_endpoint` but all the instances I found in the doc folder didn't seem to be Gloo configs but the Envoy ones, so I didn't update any of them.

# Checklist:

- [x] **N/A I think, because the changelog already talks about the snake_case to camelCase so this is just a fix.** I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] **N/A** If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] **N/A, see description for method.** I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] **Not sure besides explaining how I found these what the test would be?** I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/guides/security/auth/jwt/access_control.md [hunk 1]
```diff
@@ -9,21 +9,24 @@ The JWT feature was introduced with **Gloo Edge Enterprise**, release 0.13.16. I
 {{% /notice %}}
 
 ## Table of Contents
-- [Setup](#verifying-kubernetes-service-account-jwts)
+- [Table of Contents](#table-of-contents)
```
Comment (Rachael-Graham) [inline docs/content/guides/security/auth/jwt/access_control.md:12]: likely dont need this. this is in a few spots across these files

**Review by Rachael-Graham — COMMENTED** [review]
1 comment
Also I think you need a changelog entry

**Review by Rachael-Graham — APPROVED** [review]
LGTM

**Comment by sam-heilbron** [comment]
Thanks for doing this!

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**